keps/sig-node/20190717-seccomp-ga.md

```diff
@@ -387,9 +486,226 @@ alpha](#alternatives).
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  
+  This feature was implemented prior to the concept of feature gates. Meaning, it is already enabled 
+  in live clusters and cannot be disabled. This KEP is simply "cleaning up" the API to make it GA.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+
+  This feature (seccomp) cannot be disabled, as it is currently in a "pseudo-GA" state. 
+  However, the changes it brings are backwards compatible, and the API supports rollback 
+  of the kubernetes apiserver as described in the [Version Skew Strategy](#version-skew-strategy).
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  
+  N/A - the feature is already enabled by default since Kubernetes 1.3.
+
+* **Are there any tests for feature enablement/disablement?**
+  The e2e framework does not currently support enabling and disabling feature
+  gates. However, unit tests in each component dealing with managing data created
+  with and without the feature are necessary. At the very least, think about
+  conversion tests if API types are being modified.
+
+  N/A - the feature is already enabled by default since Kubernetes 1.3.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
+
+  The [Version Skew Strategy](#version-skew-strategy) section covers this point.
+  Running workloads should have no impact as the Kubelet will support either the 
+  existing annotations or the new fields introduced by this KEP. 
+
+* **What specific metrics should inform a rollback?**
+
+  N/A - the feature is already enabled by default since Kubernetes 1.3.
+
+* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and do that now.
+
+  Automated tests will cover the scenarios with and without the changes proposed 
+  on this KEP. As defined under [Version Skew Strategy](#version-skew-strategy),
```

Comment:
is that accurate? do we have automated upgrade/downgrade tests that exercise seccomp?

Reply:
As part of this KEP's implementation we want to create those automated tests, to ensure we are not breaking backwards compatibility until the annotations support is removed - in/after v1.23.